node-fs-readfile/cat.js

```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+
+const allFiles = process.argv;
+
+function readAll(files) {
+  const completedArray = [];
+  for (let i = 2; i < files.length; i++) {
+    fs.readFile(files[i], 'utf8', (err, data) => {
+      if (err) throw err;
+      completedArray[i] = data;
+      if (files.length === completedArray.length) {
```

Comment:
This is very very close, but a little error-prone. It is possible to instantly make the length of an array increase to a value that is misleading.

```js
var chars = [];
chars[3] = 'f';
console.log(chars.length); // 4
```

Try using a counter variable instead of the length of the completed array.